sapi/cli/php_cli.c

```diff
@@ -167,7 +167,7 @@ const opt_struct OPTIONS[] = {
 	{'S', 1, "server"},
 	{'t', 1, "docroot"},
 	{'w', 0, "strip"},
-	{'?', 0, "usage"},/* help alias (both '?' and 'usage') */
+	{'h', 0, "usage"},/* help alias (both '?' and 'usage') */
```

Comment:
Why do you *change* `?` to `h` here?

Reply:
`--usage` is an alias for `--help`, leaving this as `?` will cause `php --usage` to exit with code 1. Changing it to `h` means it will work exactly the same as `--help`

Reply:
Here I think something more needs to be done. This was actually meant as the option `php -?` which is the same as `php --usage`. But the switch then won't be able to detect invalid options... Is this question mark a special any character here?

Reply:
Right. I tested `php -?` on other shells and it didn't work, so I assumed the option was broken (E.G zsh interprets `?` differently, so you need to escape it `php -\?`).

Even with this change, `php -?` would still work but would return an exit code 1 (essentially it becomes an invalid option). I think it's better to have `usage` aliased to `help` (instead of `?`), additional handling for `php -?` can be then be added

Reply:
I don't think changing something like that is a bug fix as scripts which use this option will break by doing what should be a painless security and bug fix update.
This is more something to be done in master (i.e. PHP 8).

Reply:
Probably `php --non-existing-option` should also output some warning message like

```sh
php: unrecognized option '--non-existing-option'
Try 'php --help' for more information (or as now entire help output).
```

To not extend every single issue here to grow to numerous requests this can be probably split into more issues I guess...

Reply:
Important bit of context that took me way too long to find: `php_getopt` returns `'?'` on error.

I think given that, it is okay to use an error exit status on `?`. However, we should be doing this for all SAPIs, e.g. CGI and FPM have similar code that should be treated consistently.

Reply:
FPM already handles the exit code, so I just added the handling for CGI, although FPM uses 64 as exit code, should this be used for the other SAPIs as well or is 1 fine?

Reply:
I think 1 is fine ... as long as it's not zero, I don't think we care too much :)